File: Easy/Unique_Number_of_Occurrences/Unique_Number_of_Occurrences_v1.py
class Solution(object):
    def uniqueOccurrences(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        ans={}
        for i in arr:
            if i in ans.keys():
                ans[i]+=1
            else:
                ans[i]=1
        ans1=set()
        ans2=[]
        for i in ans.values():
            ans1.add(i)
            ans2.append(i)
        if len(ans1)==len(ans2):
            return True
        else:
            return False

File: Easy/Unique_Number_of_Occurrences/Unique_Number_of_Occurrences_v1.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
       vector<int>ans;
       int i=0;
       int size=arr.size();
       sort(arr.begin(),arr.end());
       while(i<size)
       {
        int count=1;
        for(int j=i+1;j<size;j++)
        {
            if(arr[i]==arr[j])
            {
                count++;
            }
            else
            {
                break;
            }
        }
        ans.push_back(count);
        i+=count;
       } 
       sort(ans.begin(),ans.end());
       for(int i=0;i<ans.size()-1;i++)
       {
        if(ans[i]==ans[i+1])
        return false;
        
       }
       return true;
    }
};